pkg/k8shandler/common_test.go

```diff
@@ -542,63 +543,17 @@ func buildNoCPULimitResource(cpuRequest, memLimit, memRequest resource.Quantity)
 }
 
 func areResourcesSame(lhs, rhs v1.ResourceRequirements) bool {
-
-	if !areQuantitiesSame(lhs.Limits.Cpu(), rhs.Limits.Cpu()) {
-		return false
-	}
-
-	if !areQuantitiesSame(lhs.Requests.Cpu(), rhs.Requests.Cpu()) {
-		return false
-	}
-
-	if !areQuantitiesSame(lhs.Limits.Memory(), rhs.Limits.Memory()) {
-		return false
-	}
-
-	if !areQuantitiesSame(lhs.Requests.Memory(), rhs.Requests.Memory()) {
-		return false
-	}
-
-	return true
+	return reflect.DeepEqual(lhs, rhs)
 }
 
 func areQuantitiesSame(lhs, rhs *resource.Quantity) bool {
 	return lhs.Cmp(*rhs) == 0
 }
 
 func printResource(resource v1.ResourceRequirements) string {
-	pretty := "\n{\n"
-
-	memLimit := resource.Limits.Memory()
-	memRequest := resource.Requests.Memory()
-	cpuLimit := resource.Limits.Cpu()
-	cpuRequest := resource.Requests.Cpu()
-
-	if !memRequest.IsZero() || !cpuRequest.IsZero() {
-		pretty = fmt.Sprintf("%s\tRequest:\n", pretty)
-
-		if !memRequest.IsZero() {
-			pretty = fmt.Sprintf("%s\t\tMemory: %s\n", pretty, memRequest)
-		}
-
-		if !cpuRequest.IsZero() {
-			pretty = fmt.Sprintf("%s\t\tCpu: %s\n", pretty, cpuRequest)
-		}
-	}
-
-	if !memLimit.IsZero() || !cpuLimit.IsZero() {
-		pretty = fmt.Sprintf("%s\tLimit:\n", pretty)
-
-		if !memLimit.IsZero() {
-			pretty = fmt.Sprintf("%s\t\tMemory: %s\n", pretty, memLimit)
-		}
-
-		if !cpuLimit.IsZero() {
-			pretty = fmt.Sprintf("%s\t\tCpu: %s\n", pretty, cpuLimit)
-		}
+	pretty, err := json.MarshalIndent(resource, "", "  ")
+	if err != nil {
```

Comment:
for the record... this function is only used for making the resources appear in human readable form when we need to notify that values were different. it seems odd that we'd have a potential to fail during what amounts to a `toString`...

Reply:
Agreed. but its the only way we know Serialization failed

Reply:
I think the point I was trying to make was that maybe we shouldn't have used something that could fail here.. i realize this is more streamlined than what we had before, but was there a reason to change it?

Reply:
It was changed because the previous output didn't properly convey why there was a failure; it was visually provided no difference in exp vs act.  The chance of an error here is likely small given golang is pretty good about Marshalling struct.  I would anticipate we likely never see this error.